lib/RAST_SDK/RAST_SDKImpl.pm

```diff
@@ -1476,7 +1476,7 @@ sub annotate_genomes
 	#	4. Use perl grep to see if the ref is already in the list
 	#	5. Issue a warning when a duplicate is found so user knows what happened. 
 	#
-	if (ref $genomes eq 'ARRAY') {
+	if (ref $genomes eq 'ARRAY' && scalar @{ $genomes } > 0) {
```

Comment:
A non-empty array evaluates to true, so you can write
```
if ( ref $genomes eq 'ARRAY' && @$genomes ) {
```
as a shorthand for
```
if ( ref $genomes eq 'ARRAY' && scalar @$genomes > 0 ) {
```

I would recommend using [Ref::Util](https://metacpan.org/pod/Ref::Util) for type checking as it is more elegant and less brittle than using `ref` and string comparison.

```
if ( is_arrayref( $genomes ) && @$genomes ) {
```